druid/src/widget/container.rs

```diff
@@ -154,6 +162,13 @@ impl<T: Data> Widget<T> for Container<T> {
             ctx.stroke(border_rect, &border.color.resolve(env), border_width);
         };
 
-        self.inner.paint_with_offset(ctx, data, env);
+        if self.corner_radius > 0.0 {
+            ctx.with_save(|ctx| {
+                ctx.clip(inset_panel);
+                self.inner.paint_with_offset(ctx, data, env);
+            });
+        } else {
+            self.inner.paint_with_offset(ctx, data, env);
+        }
```

Comment:
Clipping only with rounded rect seems rather inconsistent. I think having the clip always/never would be better and lead to fewer surprises.

Is clipping something we want to do here though? Is there an easy way to remove the clip from within the child's paint method? If so then this could be viewed as `overflow: hidden;` by default.

As a comparison, web CSS doesn't hide the overflow by default. Children of a rounded rect paint happily outside the bounds. With such an approach it would still be valuable to be able to opt-in to the clipping.

Perhaps the clipping should be a configuration option on `Container`?

Reply:
I think you're right about this else statement being inconsistent.